src/hooks/useActiveOrg.tsx

```diff
@@ -0,0 +1,61 @@
+import { getActiveOrgId, setActiveOrgId as setActiveOrgIdCookie } from "@propelauth/javascript"
+import { useContext, useEffect, useState } from "react"
+import { AuthContext } from "../AuthContext"
+
+const ORG_SELECTION_LOCAL_STORAGE_KEY = "__last_selected_org"
+const ACTIVE_ORG_ID_LOCAL_STORAGE_KEY = "__ACTIVE_ORG_ID"
+
+export function useActiveOrg() {
+    const context = useContext(AuthContext)
+    if (context === undefined) {
+        throw new Error("useActiveOrg must be used within an AuthProvider or RequiredAuthProvider")
+    }
+    const [activeOrgIdState, setActiveOrgIdState] = useState<string | undefined>(getActiveOrgId())
+
+    const handleLocalStorageChange = () => {
+        setActiveOrgIdState(localStorage.getItem(ACTIVE_ORG_ID_LOCAL_STORAGE_KEY) ?? undefined)
+    }
+
+    const setActiveOrgId = (orgId: string) => {
+        setActiveOrgIdCookie(orgId)
+        setActiveOrgIdState(orgId)
+        localStorage.setItem(ACTIVE_ORG_ID_LOCAL_STORAGE_KEY, orgId)
+    }
+
+    // If a cookie exists for the active org on first load, set it in local storage.
+    useEffect(() => {
+        const isLocalStorageNotSet = activeOrgIdState && !localStorage.getItem(ACTIVE_ORG_ID_LOCAL_STORAGE_KEY)
```

Comment:
I think this can lead to a confusing state where localStorage and the cookie are out of sync. It doesn't actually matter because the cookie is the source of truth, but what do you think about changing what we store in localStorage to just be a timestamp or some indication that we need to recheck the cookie.

It'll make it more clear that localStorage isn't used for anything other than the event listener.

Reply:
The idea behind this is that the cookie and the local storage are always in sync, so I'm not sure how it would lead to localStorage and the cookie being out of sync. It is really just for initial load.

Reply:
Updated to using a timestamp for the local storage instead of the ID itself